src/Factory.php

```diff
@@ -137,21 +135,28 @@ public function withoutPersisting(): self
     }
 
     /**
-     * @param callable $instantiator (array $attributes, string $class): object
+     * @internal
      */
-    final public static function registerDefaultInstantiator(callable $instantiator): void
+    final public static function boot(Manager $manager): void
     {
-        self::$defaultInstantiator = $instantiator;
+        self::$manager = $manager;
     }
 
-    final public static function registerFaker(Faker\Generator $faker): void
+    /**
+     * @internal
+     */
+    final public static function manager(): Manager
     {
-        self::$faker = $faker;
+        if (!self::$manager) {
+            throw new \RuntimeException('Factory not yet booted.'); // todo
```

Comment:
Good opportunity here for an better error message - probably your todo relates to this

Reply:
Absolutely!

Reply:
I'm going to improve all exception messages.